src/main/java/io/percy/selenium/Percy.java

```diff
@@ -164,6 +173,81 @@ public void snapshot(String name, Map<String, Object> options) {
         postSnapshot(domSnapshot, name, driver.getCurrentUrl(), options);
     }
 
+    /**
+     * Take a snapshot and upload it to Percy.
+     *
+     * @param name      The human-readable name of the screenshot. Should be unique.
+     */
+    public void screenshot(String name) throws UnsupportedOperationException {
+        Map<String, Object> options = new HashMap<String, Object>();
+        screenshot(name, options);
+    }
+
+    /**
+     * Take a snapshot and upload it to Percy.
+     *
+     * @param name      The human-readable name of the screenshot. Should be unique.
+     * @param options   Extra options
+     */
+    public void screenshot(String name, Map<String, Object> options) throws UnsupportedOperationException {
+        if (!isPercyEnabled) { return; }
+        List<String> driverArray = Arrays.asList(driver.getClass().toString().split("\\$")); // Added to handle testcase (mocked driver)
+        Iterator<String> driverIterator = driverArray.iterator();
+        String driverClass = driverIterator.next();
+        if (!driverClass.equals(RemoteWebDriver.class.toString())) { throw new UnsupportedOperationException(
+                String.format("Driver should be of type RemoteWebDriver, passed is %s", driverClass)
+        ); }
+
+        String sessionId = ((RemoteWebDriver) driver).getSessionId().toString();
+        CommandExecutor executor = ((RemoteWebDriver) driver).getCommandExecutor();
+
+        // Get HttpCommandExecutor From TracedCommandExecutor
+        if (executor instanceof TracedCommandExecutor) {
+            Class className = executor.getClass();
+            try {
+                Field field = className.getDeclaredField("delegate");
+                // make private field accessible
+                field.setAccessible(true);
+                executor = (HttpCommandExecutor)field.get(executor);
+            } catch (Exception e) {
+                log(e.toString());
+                return;
+            }
+        }
+        String remoteWebAddress = ((HttpCommandExecutor) executor).getAddressOfRemoteServer().toString();
+
+        Capabilities caps = ((RemoteWebDriver) driver).getCapabilities();
+        ConcurrentHashMap<String, String> capabilities = new ConcurrentHashMap<String, String>();
+        List<String> capsNeeded = new ArrayList<>(Arrays.asList("browserName", "platform", "version", "osVersion", "proxy"));
+        Iterator<String> iterator = capsNeeded.iterator();
+        while (iterator.hasNext()) {
+            String cap = iterator.next();
+            if (caps.getCapability(cap) != null) {
+                capabilities.put(cap, caps.getCapability(cap).toString());
+            }
+        }
```

Comment:
use enhanced for loop to club the initialization and usage statement

Reply:
I am not sure this will work, @chinmay-browserstack  check it out once. However the above code breaks if we use normal `loop` because that is not `thread safe` and hence concurrent issues will occur. Using `Iterator` makes it safe as it is `thread safe`.

Reply:
Not working with enhanced for loop: check https://github.com/percy/percy-selenium-java/pull/216/commits/1f4eb7b1c07ccd956fa1cc0391369b0b2a0eea3d